### PR TITLE
Fix ZStream#intersperse

### LINKED ISSUE #3729: ZStream#intersperse breaks with usage of Stream.repeatEffect
`ZStream#intersperse` breaks with usage of `Stream.repeatEffect`.

Test case:
```scala
testM("intersperse several from repeat effect") {
  Stream
    .repeatEffect(ZIO.succeed(42))
    .map(_.toString)
    .intersperse("@")
    .take(4)
    .runCollect
    .map(result => assert(result)(equalTo(Chunk("42", "@", "42", "@"))))
}
```
Failure: `Chunk(42,42,42,42) did not satisfy equalTo(Chunk(42,@,42,@))`

Note: Moving `ZStream#take` beforehand `ZStream#intersperse` yields the same failure.

_Edit:_ Correct failure message

### PR DESCRIPTION
**Pull request by iravid**
Resolves #3729

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -1692,11 +1692,21 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
...
+            val builder = ChunkBuilder.make[O1]()
+
+            val flagResult = os.foldLeft(flag) { (flag, o) =>
+              if (!flag) {
+                builder += o
+                !flag
+              } else {
+                builder += middle
+                builder += o
+                flag
```
Comment (regiskuckaertz) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala:1704]: Hmm 🧐 true once, true forever

Comment (luis3m) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:1704]: Noticed that, but it makes sense. You only need to avoid adding middle the first time

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:1704]: Ah let me simplify that.

Comment (luis3m) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:1704]: Maybe adding a flag var and changing the foldLeft for a foreach

**Pull request merged**